### PR TITLE
Arrays task.

### PR DESCRIPTION
**Pull request by I-am-sleeping**

### CODE REVIEW

File: arraystask/script.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+function isArray1(a) {
+  return Array.isArray(a);
```
Comment (NikitaSedyx) [inline arraystask/script.js]: this function doesn't make sense, the task is to implement your own `Array.isArray` and you did with the next 2 functions.

File: arraystask/script.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    a = 0;
+  }
+  for (var i = a; i < b; i++) {
+    result.push(i);
+  }
+  return result;
+}
+
+function compact1(a) {
+  return a.filter(Boolean);
```
Comment (NikitaSedyx) [inline arraystask/script.js:27]: it's cool if you know why it works :)

File: arraystask/script.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+function compact2(a) {
+  var truthy = [];
+  for (var i = 0; i < a.length; i++) {
+    if (Boolean(a[i])) {
```
Comment (NikitaSedyx) [inline arraystask/script.js]: will `if (a[i]) {` work?

File: arraystask/script.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+function sum2(a) {
+  var res = 0;
+  for (var i = 0; i < a.length; i++) {
+    res += a[i];
+  }
+  return res;
+}
+
+function unique(a) {
+  var uni = function(value, index, self) {
```
Comment (NikitaSedyx) [inline arraystask/script.js]: strange name `uni`. you could use `filterCb` or smth else

File: arraystask/script.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+function last(a) {
+  return a[a.length - 1];
+}
+
+function excludeLast(a, b) {
+  if ("undefined" == typeof b) {
+    b = 1;
+  }
+  for (var i = 0; i < b; i++) {
+    a.pop();
```
Comment (NikitaSedyx) [inline arraystask/script.js]: array is mutated :((

File: arraystask/script.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+function isArray2(a) {
+  return a instanceof Array;
+}
+
+function isArray3(a) {
+  return Object.prototype.toString.call(a) == '[object Array]';
+}
+
+function range(a, b) {
```
Comment (NikitaSedyx) [inline arraystask/script.js]: next cases should work:
range(10) [0..9]
range(1, 10) [1..9]
range(1, 10, 3) [1, 4, 7]
range(10, null, 3) [0, 3, 6, 9]
range(10, null) the same with range(10)
range(-2, -5) []
range(-5, -2) [-5, -4, -3]
range(-5, -2, 2) [-5, -3]
range(-5, -2, -2) []
range(-5, null, 2) []
range(-5, null, -2) [0, -2, -4]
range(-10, -20, -5) [-10, -15]
range(-20, -10, -5) []

    FYI this function can be implemented ~10 lines, firstly you should find step and length of result array (the difficult part of this function)

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]
I like some your lines

File: arraystask/script.js (hunk 3 above)
Comment (I-am-sleeping) [reply arraystask/script.js]: yes, it will work

File: arraystask/script.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
+function isArray2(a) {
+  return a instanceof Array;
+}
+
+function isArray3(a) {
+  return Object.prototype.toString.call(a) == '[object Array]';
+}
+
+function range(a, b, step) {
```
Comment (NikitaSedyx) [inline arraystask/script.js:9]: like it

File: arraystask/script.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  }
+  return a.filter(filterCb);
+}
+
+function last(a) {
+  return a[a.length - 1];
+}
+
+function excludeLast(a, b) {
+  if ("undefined" == typeof b) {
```
Comment (NikitaSedyx) [inline arraystask/script.js]: nit: you could check `b === undefined`

File: arraystask/script.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    b = 1;
+  }
+  return a.slice(0, a.length - b);
```
Comment (NikitaSedyx) [inline arraystask/script.js]: one disadvantage of this approach is when b is negative slice will return old array + [undef, undef, undef]. you can try it

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**